### ISSUE TITLE
Java.Lang.AbstractMethodError after last Xamarin.Forms update

### ISSUE THREAD

**Issue by DottorPagliaccius** [jamesmontemagno/Xamarin.Forms-PullToRefreshLayout#29]
Hi, 
PullToRefreshLayout now crashes every time I open a page the contains it.

Xamarin.Forms version 2.4.0.280
Visual Studio fo Mac 7.1.5

This is the stacktrace:
```
[MonoDroid] UNHANDLED EXCEPTION:
[MonoDroid] Java.Lang.AbstractMethodError: Exception of type 'Java.Lang.AbstractMethodError' was thrown.
[MonoDroid]   at Refractored.XamForms.PullToRefresh.Droid.PullToRefreshLayoutRenderer.get_View () [0x00005] in <0251c37903524c0f8df53a040eeb7c0a>:0 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementTracker..ctor (Xamarin.Forms.Platform.Android.IVisualElementRenderer renderer) [0x00051] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementTracker.cs:36 
[MonoDroid]   at Refractored.XamForms.PullToRefresh.Droid.PullToRefreshLayoutRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00057] in <0251c37903524c0f8df53a040eeb7c0a>:0 
[MonoDroid]   at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x0001f] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Platform.cs:285 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild (Xamarin.Forms.VisualElement view, Xamarin.Forms.Platform.Android.IVisualElementRenderer oldRenderer, Xamarin.Forms.Platform.Android.RendererPool pool, System.Boolean sameChildren) [0x00023] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:86 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x00104] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:219 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.Load () [0x00000] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:70 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetPackager (Xamarin.Forms.Platform.Android.VisualElementPackager packager) [0x00007] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:371 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000ce] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:174 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00027] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:120 
[MonoDroid]   at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x0001f] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Platform.cs:285 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild (Xamarin.Forms.VisualElement view, Xamarin.Forms.Platform.Android.IVisualElementRenderer oldRenderer, Xamarin.Forms.Platform.Android.RendererPool pool, System.Boolean sameChildren) [0x00023] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:86 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x00104] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:219 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.Load () [0x00000] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:70 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetPackager (Xamarin.Forms.Platform.Android.VisualElementPackager packager) [0x00007] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:371 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000ce] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:174 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00027] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:120 
[MonoDroid]   at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x0001f] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Platform.cs:285 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild (Xamarin.Forms.VisualElement view, Xamarin.Forms.Platform.Android.IVisualElementRenderer oldRenderer, Xamarin.Forms.Platform.Android.RendererPool pool, System.Boolean sameChildren) [0x00023] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:86 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x00104] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:219 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementPackager.Load () [0x00000] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:70 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetPackager (Xamarin.Forms.Platform.Android.VisualElementPackager packager) [0x00007] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:371 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000ce] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:174 
[MonoDroid]   at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00027] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:120 
[MonoDroid]   at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Support.V4.App.FragmentManager fragmentManager) [0x00031] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Platform.cs:322 
[MonoDroid]   at Xamarin.Forms.Platform.Android.AppCompat.FragmentContainer.OnCreateView (Android.Views.LayoutInflater inflater, Android.Views.ViewGroup container, Android.OS.Bundle savedInstanceState) [0x00008] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\AppCompat\FragmentContainer.cs:67 
[MonoDroid]   at Android.Support.V4.App.Fragment.n_OnCreateView_Landroid_view_LayoutInflater_Landroid_view_ViewGroup_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_inflater, System.IntPtr native_container, System.IntPtr native_savedInstanceState) [0x00020] in <1e89c43d3c7d48b5ba69bfc0b1bdf114>:0 
[MonoDroid]   at (wrapper dynamic-method) System.Object:387fa322-acaa-456b-99f0-2a0ca2cf9e33 (intptr,intptr,intptr,intptr,intptr)
[MonoDroid]   --- End of managed Java.Lang.AbstractMethodError stack trace ---
[MonoDroid] java.lang.AbstractMethodError
[MonoDroid] 	at md5270abb39e60627f0f200893b490a1ade.FragmentContainer.n_onCreateView(Native Method)
[MonoDroid] 	at md5270abb39e60627f0f200893b490a1ade.FragmentContainer.onCreateView(FragmentContainer.java:51)
[MonoDroid] 	at android.support.v4.app.Fragment.performCreateView(Fragment.java:2248)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1340)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1569)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1636)
[MonoDroid] 	at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:758)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2415)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2201)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl.optimizeAndExecuteOps(FragmentManager.java:2155)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2064)
[MonoDroid] 	at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:718)
[MonoDroid] 	at android.os.Handler.handleCallback(Handler.java:761)
[MonoDroid] 	at android.os.Handler.dispatchMessage(Handler.java:98)
[MonoDroid] 	at android.os.Looper.loop(Looper.java:156)
[MonoDroid] 	at android.app.ActivityThread.main(ActivityThread.java:6523)
[MonoDroid] 	at java.lang.reflect.Method.invoke(Native Method)
[MonoDroid] 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:941)
[MonoDroid] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:831)
```

**Comment by angelru** [comment]
Hello, I have the same issue.

```
09-29 02:06:36.062 I/MonoDroid( 3561): UNHANDLED EXCEPTION:
09-29 02:06:36.083 I/MonoDroid( 3561): Java.Lang.AbstractMethodError: Exception of type 'Java.Lang.AbstractMethodError' was thrown.
09-29 02:06:36.083 I/MonoDroid( 3561):   at Refractored.XamForms.PullToRefresh.Droid.PullToRefreshLayoutRenderer.get_View () [0x00005] in <0251c37903524c0f8df53a040eeb7c0a>:0 
```

This issue i have it when update a Xamarin.Forms version 2.4.0.280.

**Comment by HappyTreesDev** [comment]
I am also seeing this issue.

**Comment by Im-PJ** [comment]
I'm also seeing this error after updating to latest XF. Updated pulltorefresh to alpha version (2.0.0-alpha10) and it works.

**Comment by jamesmontemagno** [comment]
2.0 is now live and built against the latest XF and .NET Standard yay!

**Comment by DottorPagliaccius** [comment]
Thank you ;)

**Comment by khle** [comment]
My environment is at the end.

Refractored.XamForms.PullToRefresh
2.0.0.13

PullToRefresh still crashes hard. Used to be OK until I updated Xamarin.Forms to 2.5.0.122203

The accompanying sample works fine, but even if I copy the sample code over, it still crashed. I believe it's due to a compiling against a wrong lib or wrong SDK or something like that. But I checked everywhere and they all look the same. What am I still doing wrong?

Java.Lang.AbstractMethodError: 
  at java.lang.AbstractMethodError
  at at md5270abb39e60627f0f200893b490a1ade.FragmentContainer.n_onCreateView(Native Method)
  at at md5270abb39e60627f0f200893b490a1ade.FragmentContainer.onCreateView(FragmentContainer.java:51)
  at at android.support.v4.app.Fragment.performCreateView(Fragment.java:2248)
  at at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1340)
  at at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1569)
  at at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1636)
  at at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:758)
  at at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2415)
  at at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2201)
  at at android.support.v4.app.FragmentManagerImpl.optimizeAndExecuteOps(FragmentManager.java:2155)
  at at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2064)
  at at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:718)
  at at android.os.Handler.handleCallback(Handler.java:739)
  at at android.os.Handler.dispatchMessage(Handler.java:95)
  at at android.os.Looper.loop(Looper.java:148)
  at at android.app.ActivityThread.main(ActivityThread.java:5417)
  at at java.lang.reflect.Method.invoke(Native Method)
  at at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
  at at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

-------------------------------------------------------------
Visual Studio Community 2017 for Mac
Version 7.3.3 (build 5)
Installation UUID: 3d0d131d-a35c-4af4-afa6-0235381d63d1
Runtime:
	Mono 5.4.1.7 (2017-06/e66d9abbb27) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)

	Package version: 504010007

NuGet
Version: 4.3.1.4445

.NET Core
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.0.0
	1.1.1
	1.0.4
SDK: /usr/local/share/dotnet/sdk/2.0.0/Sdks
SDK Versions:
	2.0.0
	1.0.3
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.4.1/lib/mono/msbuild/15.0/bin/Sdks

Xamarin.Profiler
Version: 1.6.0
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Xamarin.Android
Version: 8.1.3.0 (Visual Studio Community)
Android SDK: /Users/obex/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.0.3 (API level 15)
		4.4   (API level 19)
		5.0   (API level 21)
		6.0   (API level 23)
		7.0   (API level 24)
		7.1   (API level 25)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 27.0.1
SDK Build Tools Version: 23.0.2

Java SDK: /usr
java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Apple Developer Tools
Xcode 9.2 (13772)
Build 9C40b

Xamarin.Mac
Version: 4.0.0.214 (Visual Studio Community)

Xamarin.iOS
Version: 11.6.1.3 (Visual Studio Community)
Hash: f70a1348
Branch: xcode9.2
Build date: 2017-12-18 14:47:16-0500

Xamarin Inspector
Version: 1.3.2
Hash: 461f09a
Branch: 1.3-release
Build date: Tue, 03 Oct 2017 18:26:57 GMT
Client compatibility: 1

Build Information
Release ID: 703030005
Git revision: b1c2982e201e71ef758866c9ade05f253a8c6f47





Build date: 2017-12-21 11:04:40-05
Xamarin addins: f397ddfbacfb39e60c9cc8d9e410f73faf8c2cbc
Build lane: monodevelop-lion-d15-5

Operating System
Mac OS X 10.13.1
Darwin 17.2.0 Darwin Kernel Version 17.2.0
    Fri Sep 29 18:27:05 PDT 2017
    root:xnu-4570.20.62~3/RELEASE_X86_64 x86_64
